catchpoint_opentsdb_bridge.py

```diff
@@ -90,26 +108,31 @@ def post(self):
-        self._tsdb.myproto.put('catchpoint.rtt.by_continent', D, 'rtt', ['nodeid', 'continent'])
-        self._tsdb.myproto.put('catchpoint.status.by_continent', D, 'counter', ['nodeid', 'continent', 'status'])
-        self._tsdb.myproto.put('catchpoint.rtt.by_city', D, 'rtt', ['nodeid', 'city'])
-        self._tsdb.myproto.put('catchpoint.status.by_city', D, 'counter', ['nodeid', 'city', 'status'])
-        self._tsdb.myproto.put('catchpoint.rtt.by_country', D, 'rtt', ['nodeid', 'country'])
-        self._tsdb.myproto.put('catchpoint.status.by_country', D, 'counter', ['nodeid', 'country', 'status'])
-        self._tsdb.myproto.put('catchpoint.rtt.by_region', D, 'rtt', ['nodeid', 'region'])
-        self._tsdb.myproto.put('catchpoint.status.by_region', D, 'counter', ['nodeid', 'region', 'status'])
-        self._tsdb.myproto.put('catchpoint.rtt.by_isp', D, 'rtt', ['nodeid', 'isp'])
-        self._tsdb.myproto.put('catchpoint.status.by_isp', D, 'counter', ['nodeid', 'isp', 'status'])
-        self._tsdb.myproto.put('catchpoint.error.by_node', D, 'counter', ['nodeid', 'error'])
-
-        defer.succeed(True)
-
+        if self._tsdb is not None and self._tsdb.myproto is not None:
+            # put each of these in tsdb, and tag the by_xxx, and nodeid so
+            # we can agg across all nodes
+            self._tsdb.myproto.put('catchpoint.rtt.by_node', D, 'rtt', ['nodeid'])
+            self._tsdb.myproto.put('catchpoint.status.by_node', D, 'counter', ['nodeid', 'status'])
+            self._tsdb.myproto.put('catchpoint.rtt.by_asn', D, 'rtt', ['nodeid', 'asn'])
+            self._tsdb.myproto.put('catchpoint.status.by_asn', D, 'counter', ['nodeid', 'asn', 'status'])
+            self._tsdb.myproto.put('catchpoint.rtt.by_continent', D, 'rtt', ['nodeid', 'continent'])
+            self._tsdb.myproto.put('catchpoint.status.by_continent', D, 'counter', ['nodeid', 'continent', 'status'])
+            self._tsdb.myproto.put('catchpoint.rtt.by_city', D, 'rtt', ['nodeid', 'city'])
+            self._tsdb.myproto.put('catchpoint.status.by_city', D, 'counter', ['nodeid', 'city', 'status'])
+            self._tsdb.myproto.put('catchpoint.rtt.by_country', D, 'rtt', ['nodeid', 'country'])
+            self._tsdb.myproto.put('catchpoint.status.by_country', D, 'counter', ['nodeid', 'country', 'status'])
+            self._tsdb.myproto.put('catchpoint.rtt.by_region', D, 'rtt', ['nodeid', 'region'])
+            self._tsdb.myproto.put('catchpoint.status.by_region', D, 'counter', ['nodeid', 'region', 'status'])
+            self._tsdb.myproto.put('catchpoint.rtt.by_isp', D, 'rtt', ['nodeid', 'isp'])
+            self._tsdb.myproto.put('catchpoint.status.by_isp', D, 'counter', ['nodeid', 'isp', 'status'])
+            self._tsdb.myproto.put('catchpoint.error.by_node', D, 'counter', ['nodeid', 'error'])
+
+            defer.succeed(True)
+        else:
+            # not connected to tsdb, dump data to file
+            with open(log_file, 'a') as outfile:
```

Comment:
should this be wrapped in a try/except block incase the file can't be opened for whatever reason?

Reply:
That would be a reasonable thing to do. I'll add a check earlier in the code to ensure that an catch / deal with the exception that'll get thrown if we can't open a handle on the log file.

Reply:
I added b21ab79 to address this.